### PR TITLE
feat(incentives):  earning events 

### PR DESCRIPTION
**Pull request by mtsitrin**
* added event for rollapp gauge distribution
* added `idToGaugeRewards []map[uint64]sdk.Coins // Per-gauge rewards for each recipient indexed by ID`
so events will have the distribution per gauge

event will look like:
```
Event: distribution
  - receiver: <address>
  - amount: <total_rewards>
  - gauge_id_1: <rewards>
  - gauge_id_2: <rewards>
  - gauge_id_3: <rewards>
  ...
```

### CODE REVIEW

**Comment by mtsitrin** [comment]
@VictorTrustyDev closes https://github.com/dymensionxyz/dymension/issues/1535?

File: x/incentives/keeper/gauge_asset.go [hunk 1]
```diff
@@ -62,46 +76,77 @@ func (d *distributionInfo) addLockRewards(owner string, rewards sdk.Coins) error
...
+	events := sdk.Events{}
+
+	for id := 0; id < len(d.idToBech32Addr); id++ {
+		attributes := []sdk.Attribute{
+			sdk.NewAttribute(types.AttributeReceiver, d.idToBech32Addr[id]),
+			sdk.NewAttribute(types.AttributeAmount, d.idToDistrCoins[id].String()),
+		}
+
+		// Add attributes for each gauge's rewards
+		for gaugeID, gaugeRewards := range d.idToGaugeRewards[id] {
```
Comment (keruch) [inline x/incentives/keeper/gauge_asset.go:100]: events should not be deterministic within nodes right? please highlight in the comment that iterating over the map is safe here

**Review by keruch — DISMISSED** [review]
LGTM! Left a few minor comments

**Review by omritoptix — DISMISSED** [review]

File: x/incentives/keeper/gauge_asset.go (hunk 1 above)
Comment (mtsitrin) [reply x/incentives/keeper/gauge_asset.go:97]: it doesn't have to be deterministic. added comment

File: x/incentives/keeper/gauge_asset.go [hunk 2]
```diff
@@ -47,10 +50,21 @@ func (k Keeper) getLocksToDistributionWithMaxDuration(ctx sdk.Context, distrTo l
...
+func (d *RewardDistributionTracker) addLockRewards(owner string, gaugeID uint64, rewards sdk.Coins) error {
 	if id, ok := d.lockOwnerAddrToID[owner]; ok {
+		// Update total rewards
 		oldDistrCoins := d.idToDistrCoins[id]
 		d.idToDistrCoins[id] = rewards.Add(oldDistrCoins...)
+
+		// Update gauge-specific rewards
+		if d.idToGaugeRewards[id] == nil {
+			d.idToGaugeRewards[id] = make(map[uint64]sdk.Coins)
+		}
```
[the remark this answers is not in the document]
Comment (mtsitrin) [reply x/incentives/keeper/gauge_asset.go]: cleared

**Pull request merged**